gpii/node_modules/transformer/src/js/Transformer.js

```diff
@@ -293,4 +293,187 @@ fluid.registerNamespace("gpii.transformer");
+
+            var hasElem = arr.indexOf(elem) !== -1;
+
+            arrayElem[transformSpec.key] = elem;
+            arrayElem[transformSpec.tag] = transformSpec.hasElemVal[hasElem ? "isTrue" : "isFalse"];
+
+            output.push(arrayElem);
+        });
+
+        return output;
+    };
+
+    /**
+     * Checks that the provied specification have the proper form, if the provided spec doesn't
+     * have the correct form, 'fluid.fail()' will be called.
+     *
+     * @param {Object} transformSpec The transform spec to be validated.
+     */
+    gpii.transformer.jsonArrayToTaggedArray.checkSpec = function (transformSpec) {
+        if (!transformSpec.tag) {
+            fluid.fail("jsonArrayToTaggedArray requires a 'tag' block set");
+        }
+        if (!transformSpec.key) {
+            fluid.fail("jsonArrayToTaggedArray requires a 'key' block set");
+        }
+        if (!transformSpec.elems || !fluid.isArrayable(transformSpec.elems)) {
+            fluid.fail("jsonArrayToTaggedArray requires a 'elems' block array");
+        }
+        if (transformSpec.hasElemVal) {
+            if (!transformSpec.hasElemVal.isTrue) {
+                fluid.fail("jsonArrayToTaggedArray 'hasElemVal' block requires a 'isTrue' value set");
+            }
+            if (!transformSpec.hasElemVal.isFalse) {
+                fluid.fail("jsonArrayToTaggedArray 'hasElemVal' block requires a 'isFalse' value set");
+            }
+        } else {
+            fluid.fail("jsonArrayToTaggedArray requires a 'hasElemVal' set");
+        }
+    };
+
+    /**
+     * Defined the inverse transform for 'gpii.transformer.jsonArrayToTaggedArray'.
+     *
+     * @param {Object} transformSpec The transform spec which inverse is going to be defined.
+     * @return {Object} The modified transform spec.
+     */
+    gpii.transformer.jsonArrayToTaggedArray.invert = function (transformSpec) {
+        transformSpec.type = "gpii.transformer.taggedArrayToJSON";
+        return transformSpec;
+    };
+
+    fluid.defaults("fluid.transforms.taggedArrayToJSON", {
+        gradeNames: "fluid.standardTransformFunction"
+    });
+
+    /**
+     * Realizes the inverse operation than 'gpii.transformer.jsonArrayToTaggedArray'. Recieves
+     * a tagged array, and returns just the values present in that array.
+     *
+     * @param {Array} arr The tagged array which values are going to be extracted.
+     * @param {Array} transformSpec The required spec for the transform. It's required
+     *      to have a key element that holds the identifier which holds the value to be extracted
+     *      from the supplied array.
+     * @return {Array} An array with the values of the provied tagged array.
+     */
+    gpii.transformer.taggedArrayToJSON = function (arr, transformSpec) {
+        var output = [];
+
+        // Check the transformSpec
+        gpii.transformer.taggedArrayToJSON.checkSpec(transformSpec);
+
+        fluid.each(arr, function (val) {
```

Comment:
The impl can be replaced with
````
var output = fluid.getMembers(arr, transformSpec.key);
````
but this inverse doesn't seem correct, since the transform ignores the value of the "tag" field and hence does no filtering of the output array.

Reply:
The problem with this transform was that I couldn't guarantee that it existed, since the elements of the provided array could not be elements of the "elems" array. I have solved it by imposing that the provided elements in the input, should be a subset of the elements provided in the "transformSpec.elems" field, this way, a inverse of the transform is always guaranteed to exist. Documentation have been updated to reflect this change.